tests/unit/test_issues.py

```diff
@@ -6,109 +6,143 @@
-                ('details', ''),
-                ('os', 'Mac OS X 13'),
-                ('problem_category', 'unknown_bug'),
-                ('submit_type', 'github-proxy-report'),
-                ('url', 'http://3139.example.com'),
-                ('username', ''), ])
-            rv = report_private_issue(form, 'http://public.example.com')
-            self.assertIsNone(rv)
-
-    @patch.object(requests, 'post')
-    def test_report_public_issue_returns_moderation_template(self, mockpost):
-        """Test the data in report_public_issue contains the right data."""
-        with app.test_request_context():
-            report_public_issue()
-            args, kwargs = mockpost.call_args
-            post_data = json.loads(kwargs['data'])
-            self.assertIs(type(post_data), dict)
-            self.assertIn('title', post_data.keys())
-            self.assertIn('body', post_data.keys())
-            self.assertIn('labels', post_data.keys())
-            self.assertEqual(['action-needsmoderation'], post_data['labels'])
-
-    def test_moderation_template(self):
-        """Check the moderation template structure.
-
-        - must be a dictionary
-        - must contain the keys: title, body
-        """
-        actual = moderation_template('ongoing')
-        self.assertIs(type(actual), dict)
-        self.assertIn('title', actual.keys())
-        self.assertIn('body', actual.keys())
-
-    def test_moderation_template_rejected(self):
-        """Check the return values are for the rejected case."""
-        actual = moderation_template('rejected')
-        self.assertEqual(actual['title'], 'Issue rejected.')
-        self.assertIn('Its original content has been deleted', actual['body'])
-
-    def test_moderation_template_ongoing(self):
-        """Check the return values are for the needsmoderation case."""
-        # test the default
-        actual = moderation_template()
-        self.assertEqual(actual['title'], 'In the moderation queue.')
-        self.assertIn('put in the moderation queue.', actual['body'])
-        # test the keyword
-        actual = moderation_template('ongoing')
-        self.assertEqual(actual['title'], 'In the moderation queue.')
-        self.assertIn('put in the moderation queue.', actual['body'])
-        # bad keyword, we go back to the default.
-        actual = moderation_template('punkcat')
-        self.assertEqual(actual['title'], 'In the moderation queue.')
-        self.assertIn('put in the moderation queue.', actual['body'])
+FORM = MultiDict([
+    ('browser', 'Firefox 99.0'),
+    ('description', 'sup'),
+    ('details', ''),
+    ('os', 'Mac OS X 13'),
+    ('problem_category', 'unknown_bug'),
+    ('submit_type', ''),
+    ('url', 'http://3139.example.com'),
+    ('username', ''), ])
+
+FORM_ANON = FORM.copy()
+FORM_ANON['submit_type'] = 'github-proxy-report'
+FORM_AUTH = FORM.copy()
+FORM_AUTH['submit_type'] = 'github-auth-report'
+
+
+@pytest.fixture
+def client():
```

Comment:
nit: maybe better to rename this method to `setup` or something now?

Reply:
good idea. done.